src/pynxtools_em/concepts/mapping_functors_pint.py

```diff
@@ -124,6 +130,76 @@ def get_case(arg):
                     return "case_five_str"
                 elif isinstance(arg[2], list):
                     return "case_five_list"
+            elif (
+                isinstance(arg[0], str)
+                and isinstance(arg[1], pint.Unit)
+                and isinstance(arg[2], str)
+                and isinstance(arg[3], str)
+            ):
+                return "case_six"
+
+
+def map_to_dtype(trg_dtype: str, value: Any) -> Any:
+    # can this be done more elegantly, i.e. written more compact?
+    # yes I already tried MAP_TO_DTYPE[trg_dtype](value) but mypy does not like it
+    # error: Argument 1 has incompatible type "generic | bool | int | float | complex |
+    #      str | bytes | memoryview"; expected "str | bytes | SupportsIndex"  [arg-type]
+    if np.shape(value) != ():
+        if trg_dtype == "u1":
+            return np.asarray(value, np.uint8)
+        elif trg_dtype == "i1":
+            return np.asarray(value, np.int8)
+        elif trg_dtype == "u2":
+            return np.asarray(value, np.uint16)
+        elif trg_dtype == "i2":
+            return np.asarray(value, np.int16)
+        # elif trg_dtype == "f2":
+        #     return np.asarray(value, np.float16)
+        elif trg_dtype == "u4":
+            return np.asarray(value, np.uint32)
+        elif trg_dtype == "i4":
+            return np.asarray(value, np.int32)
+        elif trg_dtype == "f4":
+            return np.asarray(value, np.float32)
+        elif trg_dtype == "u8":
+            return np.asarray(value, np.uint64)
+        elif trg_dtype == "i8":
+            return np.asarray(value, np.int64)
+        elif trg_dtype == "f8":
+            return np.asarray(value, np.float64)
+        elif trg_dtype == "bool":
+            if hasattr(value, "dtype"):
+                if value.dtype is bool:
+                    return np.asarray(value, bool)
+        else:
+            raise ValueError(f"map_to_dtype, hitting unexpected case for array !")
+    else:
+        if trg_dtype == "u1":
+            return np.uint8(value)
+        elif trg_dtype == "i1":
+            return np.int8(value)
+        elif trg_dtype == "u2":
+            return np.uint16(value)
+        elif trg_dtype == "i2":
+            return np.int16(value)
+        # elif trg_dtype == "f2":
+        #     return np.float16(value)
+        elif trg_dtype == "u4":
+            return np.uint32(value)
+        elif trg_dtype == "i4":
+            return np.int32(value)
+        elif trg_dtype == "f4":
+            return np.float32(value)
+        elif trg_dtype == "u8":
+            return np.uint64(value)
+        elif trg_dtype == "i8":
+            return np.int64(value)
+        elif trg_dtype == "f8":
+            return np.float64(value)
+        elif trg_dtype == "bool":
+            return try_interpret_as_boolean(value)
+        else:
+            raise ValueError(f"map_to_dtype, hitting unexpected case for scalar !")
```

Comment:
use a lookup table to avoid repetitive if-else calls
```suggestion
    mapping = {
        "u1": np.uint8,
        "i1": np.int8,
        "u2": np.uint16,
        "i2": np.int16,
        # "f2": np.np.float16,
        "u4": np.uint32,
        "i4": np.int32,
        "f4": np.int32,
        "u8": np.uint64,
        "f8": np.float64,
    } 
    if trg_dtype in mapping:
        return mapping[trg_dtype]

    elif trg_dtype == "bool":
        if np.shape(value) != ():
            if hasattr(value, "dtype"):
                if value.dtype is bool:
                    return np.asarray(value, bool)
            # Here should probably be another raise statement (or a try-except block)
        return try_interpret_as_boolean(value)
    
    else:
        if np.shape(value) != ():
            raise ValueError(f"map_to_dtype, hitting unexpected case for array !")
        raise ValueError(f"map_to_dtype, hitting unexpected case for scalar !")

Reply:
Implemented